media/com_fields/js/admin-fields-modal.js

```diff
@@ -1,30 +1,41 @@
+/**
+* PLEASE DO NOT MODIFY THIS FILE. WORK ON THE ES6 VERSION.
+* OTHERWISE YOUR CHANGES WILL BE REPLACED ON THE NEXT BUILD.
+**/
+
 /**
  * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
 Joomla = window.Joomla || {};
 
-(function(Joomla) {
-	Joomla.fieldIns = function(id, editor) {
-		/** Use the API, if editor supports it **/
-		if (window.parent.Joomla && window.parent.Joomla.editors && window.parent.Joomla.editors.instances && window.parent.Joomla.editors.instances.hasOwnProperty(editor)) {
-			window.parent.Joomla.editors.instances[editor].replaceSelection("{field " + id + "}")
-		} else {
-			window.parent.jInsertEditorText("{field " + id + "}", editor);
-		}
+(function (Joomla) {
+  'use strict';
+
+  Joomla.fieldIns = function (id, editor) {
+    // Use the API, if editor supports it
+    var win = window.parent;
+    var parentJoomla = win.Joomla;
+    if (parentJoomla && parentJoomla.editors && parentJoomla.editors.instances && parentJoomla.editors.instances.hasOwnProperty(editor)) {
+      parentJoomla.editors.instances[editor].replaceSelection('{field ' + id + '}');
+    } else {
```

Comment:
This part is deprecated in 3.x so it needs to be removed.

Also this file as a whole is wrong because the functions are called from `onclick`. We need to redo this properly without `onclick`

Reply:
Will close this until the inline scripts ahve been sorted.